src/main/java/com/cjcrafter/durability/DurabilityManager.kt

```diff
@@ -0,0 +1,79 @@
+package com.cjcrafter.armormechanics.durability
+
+import com.cjcrafter.armormechanics.ArmorMechanics
+import org.bukkit.NamespacedKey
+import org.bukkit.inventory.ItemStack
+import org.bukkit.inventory.meta.Damageable
+import org.bukkit.inventory.meta.ItemMeta
+import org.bukkit.persistence.PersistentDataType
+import java.util.List
+
+object DurabilityManager {
+    private val DURABILITY: NamespacedKey = NamespacedKey(ArmorMechanics.INSTANCE, "armor-durability")
+    private val DURABILITY_MAX: NamespacedKey = NamespacedKey(ArmorMechanics.INSTANCE, "armor-durability-max")
+    fun changeDurability(item: ItemStack, change: Int) {
+        if (getDurability(item) == -1) {
+            return
+        }
+        var durability = getDurability(item)
+        val maxDurability = getMaxDurability(item)
+        durability = Math.min(durability + change, maxDurability)
+        setDurability(item, durability)
+        val damageable = item.itemMeta
+        if (damageable !is Damageable) {
+            return
+        }
+
+        if (durability > 0) {
+
+
+
+            damageable.damage = item.type.maxDurability - (durability*item.type.maxDurability)/maxDurability
+            if (damageable.hasLore()) {
+                val lore: ArrayList<String> = ArrayList<String>(damageable.getLore())
+                for (i in lore.indices) {
+                    if (lore[i].startsWith(ArmorMechanics.DURABILITY_PRE)) {
+                        lore[i] = java.lang.String.format(ArmorMechanics.DURABILITY, durability, maxDurability)
+                    }
+                }
+                damageable.setLore(lore)
+            } else {
+                damageable.setLore(
+                    List.of(
+                        java.lang.String.format(
+                            ArmorMechanics.DURABILITY,
+                            durability,
+                            maxDurability
+                        )
+                    )
+                )
+            }
+        } else {
+            damageable.setDamage(item.getType().getMaxDurability().toInt())
+            item.setAmount(0)
+        }
+        item.setItemMeta(damageable)
+    }
+
+    fun getMaxDurability(item: ItemStack): Int {
```

Comment:
What about 1.13 support?

Reply:
Do you have a recommendation about how can it be handled? I was thinking about using NBTAPI and NBT tags but I'm not sure if it is the best way to do that & would you be okay with adding NBTAPI as a soft dependency for this feature etc

Reply:
`CompatibilityAPI.getNbtCompatibility()`

Reply:
> What about 1.13 support?

Also plugins's JVM target is Java 16, how does the plugin aim 1.13 support (1.13 Spigot and Paper supports up to Java 12) with that?

Reply:
You can run MC 1.12.2 on Java 16

Reply:
Not sure about you can run Minecraft **client** 1.12 with Java 16 (and not sure what does it affect) but you can't run Spigot nor Paper 1.12 **server** with Java 16

Reply:
I mean... I just did it now. And I have been doing it for years.